apps/election-manager/src/screens/ManualDataImportScreen.tsx

```diff
@@ -0,0 +1,371 @@
+
+import {
+  ExternalTallySourceType,
+  TallyCategory,
+  VotingMethod,
+} from '../config/types'
+import routerPaths from '../routerPaths'
+
+import AppContext from '../contexts/AppContext'
+
+import Prose from '../components/Prose'
+import Table, { TD } from '../components/Table'
+
+import NavigationScreen from '../components/NavigationScreen'
+import Button, { SegmentedButton } from '../components/Button'
+import Select from '../components/Select'
+import {
+  expandEitherNeitherContests,
+  getContestsForPrecinct,
+  getAllPossibleCandidatesForCandidateContest,
+} from '../utils/election'
+import TextInput from '../components/TextInput'
+import {
+  convertTalliesByPrecinctToFullExternalTally,
+  getEmptyExternalTalliesByPrecinct,
+  getEmptyExternalTally,
+  getTotalNumberOfBallots,
+} from '../utils/externalTallies'
+
+const MANUAL_DATA_NAME = 'Manually Added Data'
+
+const TallyInput = styled(TextInput)`
+  width: 4em;
+`
+
+const ContestData = styled.div`
+  padding: 10px 0;
+`
+
+const ManualDataImportScreen: React.FC = () => {
+  const {
+    electionDefinition,
+    fullElectionExternalTallies,
+    saveExternalTallies,
+  } = useContext(AppContext)
+  const { election } = electionDefinition!
+  const history = useHistory()
+
+  const existingManualDataTallies = fullElectionExternalTallies.filter(
+    (t) => t.source === ExternalTallySourceType.Manual
+  )
+  const existingManualData =
+    existingManualDataTallies.length === 1
+      ? existingManualDataTallies[0]
+      : undefined
+  const existingTalliesByPrecinct = existingManualData?.resultsByCategory.get(
+    TallyCategory.Precinct
+  )
+  const [talliesByPrecinct, setTalliesByPrecinct] = useState(
+    existingTalliesByPrecinct ?? getEmptyExternalTalliesByPrecinct(election)
+  )
+  const [ballotType, setBallotType] = useState<VotingMethod>(
+    existingManualData?.votingMethod ?? VotingMethod.Precinct
+  )
+
+  const initialPrecinct = election.precincts[0]?.id
+  const [currentPrecinctId, setCurrentPrecinctId] = useState(initialPrecinct)
+  const currentPrecinct = election.precincts.find(
+    (p) => p.id === currentPrecinctId
+  )
+  const currentPrecinctTally =
+    talliesByPrecinct[currentPrecinctId] ?? getEmptyExternalTally()
+
+  const handleImportingData = async () => {
+    // Turn the precinct tallies into a CSV SEMS file
+    // Save that file as the external results file with a name implied manual data entry happened
+    const externalTally = convertTalliesByPrecinctToFullExternalTally(
+      talliesByPrecinct,
+      election,
+      ballotType,
+      ExternalTallySourceType.Manual,
+      MANUAL_DATA_NAME,
+      new Date()
+    )
+    // Don't modify any external tallies for non-manual data
+    const newTallies = fullElectionExternalTallies.filter(
+      (t) => t.source !== ExternalTallySourceType.Manual
+    )
+    // Add the new tally
+    newTallies.push(externalTally)
+    await saveExternalTallies(newTallies)
+    history.push(routerPaths.tally)
```

Comment:
any particular reason we're trying to track history here? I don't think we're supporting the back button anywhere, so wondering if we need to track it now.

Reply:
Not sure I understand what you mean, this is just redirecting the user back to the main tally page after they finish saving the manual data.